django_project/frontend/tests/test_metrics.py

```diff
@@ -141,3 +141,30 @@ def test_total_count_per_activity(self):
             response.data[0]['activities'][0],
             activity_total,
         )
+
+
+class SpeciesPopulationTotalAndDensityTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.url = reverse("species_population_total_density")
+
+    def test_species_population_total_and_density(self):
+        url = self.url
+        response = self.client.get(url, **self.auth_headers)
+        owned_species = self.owned_species[0]
+        population = AnnualPopulation.objects.filter(
```

Comment:
Hi @ketankartoza , test shouldn't replicate the exact logic of the function being tested. 
Instead, it should confirm that the function behaves as expected given certain inputs and scenarios.

It would be better to compare the output of your function with predetermined values.

For example : 

```
self.assertEqual(
            response.data[0]['density'].get('density'), 100
        )
        self.assertEqual(
            response.data[0]['density'].get('total'), 50
        )
```

Reply:
Okay @dimasciput, I will write tests as you suggested.